resharper/resharper-unity/src/UnityApi.cs

```diff
@@ -129,22 +129,22 @@ public bool IsInjectedField([CanBeNull] IField field)
         // scene. We don't have enough information to do this by name, so we'll mark all potential event handlers as
         // implicitly used by Unity
         // See https://github.com/Unity-Technologies/UnityCsReference/blob/02f8e8ca594f156dd6b2088ad89451143ca1b87e/Editor/Mono/Inspector/UnityEventDrawer.cs#L397
-        public bool IsPotentialEventHandler([CanBeNull] IMethod method)
+        public bool IsPotentialEventHandler([CanBeNull] IMethod method, bool isFindUsages = false)
         {
             if (method == null || !method.ReturnType.IsVoid())
                 return false;
 
             // Type.GetMethods() returns public instance methods only
-            if (method.GetAccessRights() != AccessRights.PUBLIC || method.IsStatic)
+            if (method.GetAccessRights() != AccessRights.PUBLIC && !isFindUsages|| method.IsStatic)
```

Comment:
I think it's worth adding a comment here to say that the Unity Editor will only list public methods, but will invoke any method, even if it's private.

It might even be worth getting rid of the `isFindUsages`, and making `true` the default action.